### ISSUE TITLE
Java 8 Bootstrap

### ISSUE THREAD

**Issue by Downchuck** [awslabs/emr-bootstrap-actions#131]
Newer versions of components such as Presto may require Java 8, a stock bootstrap action would make it easier to install them.

There is an action here, but it's not the right one:
https://gist.github.com/ericeijkelenboom/9951500

**Comment by Downchuck** [comment]
Hive metastore start fails:
```
Unable to determine Hadoop version information.
'hadoop version' returned:
```

Likely due to some very loose parsing of hadoop version output and a warning message:
```
[hadoop@ip-x 3]$ hadoop version
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=128m; support was removed in 8.0
Hadoop 2.4.0-amzn-5
Subversion http://aws.amazon.com/elasticmapreduce/ -r b794955381db8469b9864edc04e108e698c2d897
Compiled by Elastic MapReduce on 2015-05-22T23:40Z
Compiled with protoc 2.5.0
From source with checksum 6c725ed23b3ecb95921fe461587fccf
```

**Comment by Downchuck** [comment]
This seems to work, in a nutshell (ruby style, following presto bootstrap):

```rb
  println "copyToLocal from #{s3_path_to_jdk_18_bin}"
  run "hdfs dfs -copyToLocal #{s3_path_to_jdk_18_bin} /tmp/jdk-8u45-linux-x64.rpm"
  sudo "#{$install_cmd} /tmp/jdk-8u45-linux-x64.rpm"
  sudo "update-alternatives --install /usr/bin/java java /usr/java/jdk1.8.0_45/bin/java 200000"
  sudo "update-alternatives --install /usr/bin/javac javac /usr/java/jdk1.8.0_45/bin/javac 200000"
  sudo "update-alternatives --set java /usr/java/jdk1.8.0_45/bin/java"
  sudo "update-alternatives --set javac /usr/java/jdk1.8.0_45/bin/javac"

  #remove ignored MaxPermSize setting
  sudo "sed -i  -- 's/ -XX:MaxPermSize=128m//g' /home/hadoop/etc/hadoop/*.sh"
```

**Comment by shenzhun** [comment]
Hi Downchuck,

Did you successfully install presto 0.99 on AWS ?  I followed https://github.com/awslabs/emr-bootstrap-actions/tree/master/presto/latest, but the java version was still 1.7.

**Comment by Downchuck** [comment]
I did not end up using latest; I tweaked an older branch prior to noticing that folder. Additionally, presto is up around 0.109 now... I am waiting on a significant fix in its ORC support before diving in again.

**Comment by mombergm** [comment]
Hi Guys. 
We will be updating the Presto Bootstrap to facilitate the latest version on Tuesday.  Please check back then for all the changes that we added as well as the latest release of Presto

**Comment by Downchuck** [comment]
Here's that big issue, I referred to:
https://github.com/facebook/presto/issues/3023

Most subselect or IN query clauses will result in the entire table being scanned.

**Comment by Downchuck** [comment]
Presto issue fixed:
https://github.com/facebook/presto/issues/3023#event-362447199

**Comment by electrum** [comment]
That fix will be available in the next release, 0.112.

**Comment by Downchuck** [comment]
@mombergm It makes more sense to support 0.112 as a default given the fix to ORC handling. I'm going to close out this issue, as it seems Presto is the only project pressing for Java8 bootstrap.
